src/tools/rbd/action/Import.cc

```diff
@@ -594,6 +618,16 @@ static int do_import_header(int fd, int import_format, uint64_t &size, librbd::I
       r = decode_and_set_image_option(fd, RBD_IMAGE_OPTION_STRIPE_UNIT, opts);
     } else if (tag == RBD_EXPORT_IMAGE_STRIPE_COUNT) {
       r = decode_and_set_image_option(fd, RBD_IMAGE_OPTION_STRIPE_COUNT, opts);
+    } else if (tag == RBD_EXPORT_IMAGE_META) {
+      r = utils::read_string(fd, length, &key);
```

Comment:
Nit: why not follow the existing pattern of just calling a helper function to read the associated values of the tag struct and update the image (avoiding the intermediate list)?

Reply:
we havn't create an import_image yet. so what we could do is just to collect them all and set image-meta later.